charts/linkerd2/templates/controller.yaml

```diff
@@ -100,7 +100,11 @@ spec:
         volumeMounts:
         - mountPath: /var/run/linkerd/config
           name: config
-      - {{- include "partials.proxy" . | indent 8 | trimPrefix (repeat 7 " ") }}
+      {{- $tree := deepCopy . }}      
+      {{- if not (empty .Values.publicAPIProxyResources) }}
+      {{- $r := merge .Values.publicAPIProxyResources .Values.global.proxy.resources }}
+      {{- $_ := set $tree.Values.global.proxy "resources" $r }}
```

Comment:
There's a missing `{{- end }}` here that is preventing the chart to render properly.